### PR TITLE
Fix wrong type for comment key

### LINKED ISSUE #2147: User is blocked to proceed for Transaction and Exchange.
**Frequency:** 100%
**App version:** Staging build v1.5.2
**Repro on:** Samsung Galaxy Note 5 (7.0), Samsung Galaxy s5 (6.0.1)  
**Testing Account** 518 468 2551/[BackupKay ](https://docs.google.com/spreadsheets/d/1Iy41GGA6e2pES2XHjBn7nb8X3n8zZsu7EW1jvwD2e70/edit#gid=0&range=A3)

**Repro Steps:** 
1) Launch the app. 
2) Tap + tab. 
3) Select celo contact user. 
4) proceed for transaction and tap on send button. 
5) Observe “Failure Sending Payment” message is shown 

**Impact:** User is blocked to send money and exchange the cUSD to cGLD. 
**Investigation:**
1. “Failure Sending Payment” message is shown when user tap on Send button. 
2. Sometime Confirming… text is shown on “Payment Sent” transaction while proceed for transaction and exchange
3. User is able to send request payment. 
4. Issue is always repro either celo CLI toggle is enabled or disabled 

**Current Behavior:** Confirming… text is shown on “Payment Sent” transaction while proceed for transaction and exchange
**Expected Behavior:** User should be able to send the payment and exchange the cUSD to cGLD. 
**Attachment:** [Blocked_Transaction&Exchange.mp4](https://drive.google.com/file/d/1WzQCsHczZap9LdlzSz6H2D6xzOj0wQH2/view?usp=sharing) (Bug observe at  13 sec)

### PR DESCRIPTION
**Pull request by i1skn**
### Description

Fix wrong type for comment key

### Tested
Android

### Other changes

Fix the same issue in verification flow

### Related issues

- Fixes https://github.com/celo-org/celo-monorepo/issues/2147

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/identity/commentKey.ts [hunk 1]
```diff
@@ -5,8 +5,8 @@ import { contractKit } from 'src/web3/contracts'
 
 export async function getCommentKey(address: string): Promise<Buffer | null> {
   const accountsWrapper: AccountsWrapper = await contractKit.contracts.getAccounts()
-
-  const hexString = (await accountsWrapper.getDataEncryptionKey(address)).join()
+  // getDataEncryptionKey actually returns a string instead of an array
+  const hexString = [await accountsWrapper.getDataEncryptionKey(address)].join()
```
Comment (annakaz) [inline packages/mobile/src/identity/commentKey.ts:9]: Could you create an issue to change the Account `getDataEncryptionKey` type signature in contractkit?

File: packages/mobile/src/identity/verification.ts [hunk 2]
```diff
@@ -522,7 +522,10 @@ async function isAccountUpToDate(
     accountsWrapper.getDataEncryptionKey(address),
   ])
   return (
-    eqAddress(currentWalletAddress, address) && currentDEK && eqAddress(currentDEK.join(), dataKey)
+    // currentDEK is actually a string instead of an array
+    eqAddress(currentWalletAddress, address) &&
+    currentDEK &&
+    eqAddress([currentDEK].join(), dataKey)
```
Comment (annakaz) [inline packages/mobile/src/identity/verification.ts]: Not sure, but is it possible to fix the type when `getDataEncryptionKey` is called in line 522 without losing parallelization? Would be nice to be consistent with the other change (and to make it easy to search `getDataEncryptionKey` once the type signature in contractkit is fixed)

**Review by annakaz — APPROVED** [review]
Great job catching this 💪 left some minor comments

**Review by jmrossy — APPROVED** [review]

File: packages/mobile/src/identity/commentKey.ts (hunk 1 above)
Comment (i1skn) [reply packages/mobile/src/identity/commentKey.ts:9]: Done - https://github.com/celo-org/celo-monorepo/issues/2265

**Comment by i1skn** [comment]
@annakaz PTAL - I've addressed the feedback.

**Review by annakaz — APPROVED** [review]

**Pull request merged**